dlt/common/runners/stdout.py

```diff
@@ -24,6 +25,45 @@ def exec_to_stdout(f: AnyFun) -> Iterator[Any]:
             print(encode_obj(rv), flush=True)
 
 
+def iter_std(venv: Venv, command: str, *script_args: Any) -> Iterator[Tuple[int, str]]:
+    """Starts a process `command` with `script_args` in environment `venv` and returns iterator
+    of (filno, line) tuples where `fileno` is 1 for stdout and 2 for stderr. `line` is
+    a content of a line with stripped new line character.
+
+    Use -u in scripts_args for unbuffered python execution
+    """
+    with venv.start_command(
+        command, *script_args, stdout=PIPE, stderr=PIPE, bufsize=1, text=True
+    ) as process:
+        exit_code: int = None
+        q_: queue.Queue[Tuple[int, str]] = queue.Queue()
+
+        def _r_q(std_: int) -> None:
+            stream_ = process.stderr if std_ == 2 else process.stdout
+            for line in iter(stream_.readline, ""):
+                q_.put((std_, line[:-1] if line[-1] == "\n" else line))
+            # close queue
+            q_.put(None)
+
+        # read stderr with a thread, selectors do not work on windows
+        t_out = Thread(target=_r_q, args=(1,), daemon=True)
+        t_out.start()
+        t_err = Thread(target=_r_q, args=(2,), daemon=True)
+        t_err.start()
+        while line := q_.get():
```

Comment:
Should we have some `hard` timeout after it that we shut down or do we just let users use `Ctrl+C`?

Reply:
not now I think. adding timeout is trivial if we need it